City-of-Helsinki/drupal-helfi-platform-config
# UHF-6666: Search filters

# [UHF-6666: Search filters](https://helsinkisolutionoffice.atlassian.net/browse/UHF-6666)
Added filters to LinkedEvents React-version

## What was done
* Added filters according to specs
* Filters rendered conditionally based on paragraph choices
* Selecting stuff from filters and submitting creates an API call to LinkedEvents and renders results

## How to install
* Get any instance up and running
* Get this branch: `composer require drupal/helfi_platform_config:dev-UHF-6666-search-filters`
* Get the style branch for HDBT: `composer require drupal/hdbt:dev-UHF-3128-sote-linkedevents-filtering`
* Run `make drush-updb`
* Run `make drush-cr`

## How to test
* Create or edit a landing- or standard page. 
* Add an `Events` paragraph. Fill fields.
  * API URL = ` https://tapahtumat.hel.fi/fi/events?text=jooga`
* Visit the page. 
* Test the filters: 
  * Changing filter values shouldn't execute search, only clicking the submit button
  * Test that filters work as expected
* "Lataa lisää" will be removed.  At this point, only 5 first events will be shown, regadless of result count. This will be handled later with Pagination component.

## Other PRs
* https://github.com/City-of-Helsinki/drupal-hdbt/pull/424
